dcc-storage-server/src/main/java/org/icgc/dcc/storage/server/authorize/BasicScopeAuthorizationStrategy.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2016 The Ontario Institute for Cancer Research. All rights reserved.                             
+ *                                                                                                               
+ * This program and the accompanying materials are made available under the terms of the GNU Public License v3.0.
+ * You should have received a copy of the GNU General Public License along with                                  
+ * this program. If not, see <http://www.gnu.org/licenses/>.                                                     
+ *                                                                                                               
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY                           
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES                          
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT                           
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,                                
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED                          
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;                               
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER                              
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN                         
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.icgc.dcc.storage.server.authorize;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.NonNull;
+import lombok.val;
+import lombok.extern.slf4j.Slf4j;
+
+import org.icgc.dcc.storage.server.exception.NotRetryableException;
+import org.icgc.dcc.storage.server.util.Access;
+import org.icgc.dcc.storage.server.util.AuthScope;
+import org.springframework.beans.factory.annotation.Value;
+
+@Slf4j
+public class BasicScopeAuthorizationStrategy extends AbstractScopeAuthorizationStrategy {
+
+  @Value("${auth.server.downloadScope}")
+  private String downloadScope;
+
+  // @Autowired
+  // private MetadataService metadataClient;
+
+  public BasicScopeAuthorizationStrategy() {
+    super();
+  }
+
+  BasicScopeAuthorizationStrategy(final String scopeStr) {
+    super(scopeStr);
+  }
+
+  @Override
+  protected void setAuthorizeScope(String scopeStr) {
+    downloadScope = scopeStr;
```

Comment:
Again, not thread safe. Is there any reason we cannot make these classes immutable by design? At the very least it should have prototype or request scope, but I'm not sure what Spring Security expects in this context.